src/components/DraggableList/PDraggableList.vue

```diff
@@ -0,0 +1,257 @@
+<template>
+  <div class="p-draggable-list" :class="classes.root">
+    <template v-for="(item, i) in modelValue" :key="i">
+      <div
+        ref="items"
+        class="p-draggable-list__item"
+        :class="classes.item(i)"
+        :draggable="itemIsDraggable(i)"
+        tabindex="0"
+        @dragover="handleDragOver($event, i)"
+        @dragleave="handleDragLeave(i)"
+        @dragstart="handleDragStart"
+        @dragend="handleDragEnd"
+        @drop="drop"
+        @keydown="handleKeydown($event, i)"
+      >
+        <div class="p-draggable-list__item-handle" @mousedown="handleMouseDown(i)" @mouseup="handleMouseUp">
+          <slot name="handle" v-bind="{ item, index: i }">
+            <PIcon icon="DragHandle" />
+          </slot>
+        </div>
+        <div class="p-draggable-list__item-content">
+          <slot v-bind="{ item, index: i }" />
```

Comment:
What do you think about binding a delete callback as part of the scope? That way at implementation the logic for deleting doesn't need to be written there as well. Can just grab it from the scope.

Reply:
Love that - I'll think through if there are other pieces we need to emit from the top level component and/or expose through the scope. At the moment I'm not sure about the level of customization this is giving us for using this on individual fields

Reply:
I think exposing some of the move item functionality as callbacks would be nice as well. Things like `moveUp`, `moveDown`, `moveToTop`, and `moveToBottom` that way the implementation could add buttons or menu items for that functionality as well.